packages/extension-commerce-bm-seo/src/setup-app.ts

```diff
@@ -0,0 +1,166 @@
+// `withApplicationExtensionStore` usage below as well.
+const sliceInitializer: SliceInitializer<StoreSlice> = (set) => ({
+    count: 0,
+    increment: () => set((state) => ({count: state.count + 1})),
+    decrement: () => set((state) => ({count: state.count - 1}))
+})
+
+class CommerceBmSeo extends ApplicationExtension<Config> {
+    static readonly id = extensionMeta.id
+
+    /**
+     * Use this method to wrap or enhance your PWA-Kit application using [React higher-order components](https://legacy.reactjs.org/docs/higher-order-components.html).
+     * You can use this to add visual treatments to your application, change the props that are supplied to the application component
+     * or add things like providers and contexts to be used throughout your app.
+     */
+    extendApp<T extends React.ComponentType<T>>(
+        App: React.ComponentType<T>
+    ): React.ComponentType<T> {
+        const HOCs = [
+            // Example higher-order component, this can be safely removed.
+            sampleHOC,
+            // Optionally include state for this extension using `withApplicationExtensionStore`
+            (component: React.ComponentType<any>) =>
+                withApplicationExtensionStore(component, {
+                    id: extensionMeta.id,
+                    initializer: sliceInitializer
+                })
+        ]
+
+        return applyHOCs(App, HOCs)
+    }
+
+    /**
+     * TODO: update this comment
+     * This method is used to make changes to the PWA-Kit application routes. If your extension adds a new page to the application
+     * then you can add it to the router here. The routes passed to this method is an accrued list of routes that have been added
+     * from extensions applied before it. It is called during the `getRoutes` phase on both the server and the client.
+     *
+     * NOTE: If you instead want to modify a list of all the routes, refer to the `beforeRouteMatch` below.
+     */
+    async getRoutesAsync({locals}: GetRoutesParams): Promise<RouteProps[]> {
+        // TODO: do we need this?
+        if (!locals.originalUrl) {
+            return Promise.resolve([])
+        }
+
+        // Make SEO GET Url Mapping API call
+        const config = this.getConfig()
+        const shopperSeo = await getShopperSeoClient(locals, config)
+        const urlMapping = await shopperSeo.getUrlMapping({
+            parameters: {urlSegment: locals.originalUrl}
+        })
+        console.log('--- url mapping', urlMapping)
+
+        if (!urlMapping.resourceType) {
+            return Promise.resolve([])
+        }
+
+        // Transform URL mapping to serialized route
+        const requestURL = new URL(locals.originalUrl, getAppOrigin(locals))
+        const componentName =
+            config.resourceTypeToComponentMap[urlMapping.resourceType as keyof typeof config.resourceTypeToComponentMap]
+        
+        // Create a placeholder component so it can be deserialized later in beforeRouteMatch
```

Comment:
Yes, but.. I think more importantly, we need a placeholder component because we don't have access to the component until later. This component is in a separate extension, that's why.